language/code/code_retrieval_augmented_generation.ipynb

```diff
@@ -1,785 +1,773 @@
-   "metadata": {
-    "colab": {
-     "base_uri": "https://localhost:8080/"
-    },
-    "id": "1svTVwtBS0zP",
-    "outputId": "234df47d-a693-4790-ac39-853e8882724d"
-   },
-   "outputs": [],
-   "source": [
-    "response = code_llm.predict(text=user_question, max_output_tokens=2048, temperature=0.1)\n",
-    "print(response)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "id": "JPm8qdxzwPM0"
-   },
-   "source": [
-    "### 6. Run prompt using RAG Chain & compare results\n",
-    "To generate response we use code-bison however can also use code-gecko and codechat-bison"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
-    "colab": {
-     "base_uri": "https://localhost:8080/"
-    },
-    "id": "ZMz3nPMyVoj_",
-    "outputId": "6ea6b550-365e-4147-c42a-7dcab4dde038"
-   },
-   "outputs": [],
-   "source": [
-    "results = qa_chain({\"query\": user_question})\n",
-    "print(results[\"result\"])"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "id": "HF3lVWK1wjxe"
-   },
-   "source": [
-    "### Let's try another prompt"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
-    "colab": {
-     "base_uri": "https://localhost:8080/"
-    },
-    "id": "jel0ON68XiU7",
-    "outputId": "cf6d8a16-4e03-4611-88ab-29be2bdde4e5"
-   },
-   "outputs": [],
-   "source": [
-    "user_question = \"Create python function that takes text input and returns embeddings using Langchain with VertexAI textembedding-gecko model\"\n",
-    "\n",
-    "\n",
-    "response = code_llm.predict(text=user_question, max_output_tokens=2048, temperature=0.1)\n",
-    "print(response)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
+  "cells": [
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "5norOZI0mA6s"
+      },
+      "outputs": [],
+      "source": [
+        "# Copyright 2023 Google LLC\n",
+        "#\n",
+        "# Licensed under the Apache License, Version 2.0 (the \"License\");\n",
+        "# you may not use this file except in compliance with the License.\n",
+        "# You may obtain a copy of the License at\n",
+        "#\n",
+        "#     https://www.apache.org/licenses/LICENSE-2.0\n",
+        "#\n",
+        "# Unless required by applicable law or agreed to in writing, software\n",
+        "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+        "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+        "# See the License for the specific language governing permissions and\n",
+        "# limitations under the License."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "XNPE46X8mJj4"
+      },
+      "source": [
+        "# Use Retrieval Augmented Generation (RAG) with Codey API's\n",
+        "\n",
+        "<table align=\"left\">\n",
```

Comment:
This is rendering oddly in nbreview, not sure if it's just a nbreview rendering issue, or your table seems to be missing `<tr>` tags?

![Screenshot 2024-02-22 at 11 46 22 AM](https://github.com/GoogleCloudPlatform/generative-ai/assets/303236/4e279589-d312-4ea9-b5d8-46e99853292a)

Reply:
That's just an NBReview issue